### PR TITLE
DEV-1101 Integrate pipeline5 with bcl2fastq on GCP

### PR DESCRIPTION
**Pull request by pauldwolfe**
Adds a toggle that once enabled, will upload fastq input from GCP rather than the SBP S3.
The location of the fastq is still taken from the SBP API (run->sample->fastq).

Also updated the Bcl2Fastq command thread settings to match production.

### CODE REVIEW

File: bcl2fastq/src/main/java/com/hartwig/bcl2fastq/metadata/SbpFastqMetadataApi.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public SbpSample findOrCreate(String barcode, String submission) {
                 try {
                     Response response = samples().request()
                             .post(Entity.entity(objectMapper.writeValueAsString(sample), MediaType.APPLICATION_JSON_TYPE));
-                    if (response.getStatus() == 200) {
+                    if (response.getStatus() == 201) {
```
Comment (nedleitch) [inline bcl2fastq/src/main/java/com/hartwig/bcl2fastq/metadata/SbpFastqMetadataApi.java]: `Response.Status.CREATED` or something similar?

File: cluster/src/main/java/com/hartwig/pipeline/execution/vm/VirtualMachinePerformanceProfile.java [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
-public interface VirtualMachinePerformanceProfile extends PerformanceProfile{
+public interface VirtualMachinePerformanceProfile extends PerformanceProfile {
+
+    @Value.Default
+    default String uri() {
+        return machineType().uri();
+    }
 
-    String uri();
+    MachineType machineType();
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/execution/vm/VirtualMachinePerformanceProfile.java]: Much better.

File: bcl2fastq/src/main/java/com/hartwig/bcl2fastq/conversion/ResultAggregation.java [hunk 3]
```diff
@@ -95,7 +95,15 @@ static ConvertedFastq fastq(IlluminaSample sample, Map.Entry<String, List<Blob>>
     }
 
     private static String outputPath(final IlluminaSample sample, final Stats stats, final Blob blobR1) {
-        return stats.flowcell() + "/" + sample.barcode() + "/" + new File(blobR1.getName()).getName();
+        String[] fileNameSplit = new File(blobR1.getName()).getName().split("_");
```
Comment (nedleitch) [inline bcl2fastq/src/main/java/com/hartwig/bcl2fastq/conversion/ResultAggregation.java]: We have had surprises in the past, would it be wise to check the filename conforms to expectations?

File: bcl2fastq/src/main/java/com/hartwig/bcl2fastq/metadata/SbpFastqMetadataApi.java (hunk 1 above)
Comment (pauldwolfe) [reply bcl2fastq/src/main/java/com/hartwig/bcl2fastq/metadata/SbpFastqMetadataApi.java]: Good suggestion

File: bcl2fastq/src/main/java/com/hartwig/bcl2fastq/conversion/ResultAggregation.java (hunk 3 above)
Comment (pauldwolfe) [reply bcl2fastq/src/main/java/com/hartwig/bcl2fastq/conversion/ResultAggregation.java]: hmm, as this file convention is produced by the version of bcl2fastq also in this codebase I think its fine to have this implicit dependency on the filename.

**Pull request merged**